### ISSUE TITLE
[MuPDF] Fulltext search ignores leading and trailing spaces

### ISSUE THREAD

**Issue by jonnyl2** [koreader/koreader#11029]
The MuPDF Fulltext search engine will ignore any leading or trailing spaces, so a search for " an", "an ", or " an " will for example all yield the following results:

![image](https://github.com/koreader/koreader/assets/95502269/d665fea6-c09d-4e1b-a97f-799f960014b3)

As regex-search is not available on fixed-layout docs, those spaces are helpful to limit searches to whole words. (On CRE Fulltext search, spaces are respected even when regex search is disabled.)

**Comment by ryanwwest** [comment]
I agree that this would be nice to have. Another nice option might be search for whole words only. I think regex could do this but it's quite a common option and a checkbox would be more convenient.
